packages/babel-parser/src/parser/statement.js

```diff
@@ -366,13 +366,20 @@ export default class StatementParser extends ExpressionParser {
     return this.finishNode(node, "DebuggerStatement");
   }
 
+  parseHeaderExpression(): N.Expression {
+    this.expect(tt.parenL);
+    const val = this.parseExpression();
+    this.expect(tt.parenR);
+    return val;
+  }
```

Comment:
What is the purpose of this? This is an exact duplicate of `parseParenExpression()`

Reply:
`parse*Expression()` should not be in `statement.js`, as it is about expressions.

I also cannot fully understand the idea behind that name, what does `Header` mean in this context?
`parseParenExpression`, was pretty clear to me -> an expression inside of parens.

Reply:
It is only used to parse statements headers.
I like the new name, since with the old name it can be confused with something which produces a `ParenthesizedExpression`.

Reply:
Ok I see.

Reply:
That was exactly my point. When I first started looking into this I was expecting parseParenExpression to return a ParenExpression expression node. Later I learned that it is called ParenthesizedExpression and that the parser never produced these. This was my attempt at cheating this.

Another name could be parseControlFlowHeaderParenthesizedExpression or some combination of that.